src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.OneNote/Main.cs

```diff
@@ -0,0 +1,150 @@
+    {
+        /// <summary>
+        /// A value indicating if the OneNote interop library was able to successfully initialize.
+        /// </summary>
+        private readonly bool _oneNoteInstalled;
+
+        /// <summary>
+        /// The initial context for this plugin (contains API and meta-data)
+        /// </summary>
+        private PluginInitContext? _context;
+
+        /// <summary>
+        /// The path to the icon for each result
+        /// </summary>
+        private string _iconPath;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Main"/> class.
+        /// </summary>
+        public Main()
+        {
+            UpdateIconPath(Theme.Light);
+            try
+            {
+                _ = OneNoteProvider.PageItems.Any();
+                _oneNoteInstalled = true;
+            }
+            catch (COMException)
+            {
+                // OneNote isn't installed, plugin won't do anything.
+                _oneNoteInstalled = false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the localized name.
+        /// </summary>
+        public string Name => Resources.PluginTitle;
+
+        /// <summary>
+        /// Gets the localized description.
+        /// </summary>
+        public string Description => Resources.PluginDescription;
+
+        /// <summary>
+        /// Initialize the plugin with the given <see cref="PluginInitContext"/>
+        /// </summary>
+        /// <param name="context">The <see cref="PluginInitContext"/> for this plugin</param>
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+
+            _context.API.ThemeChanged += OnThemeChanged;
+            UpdateIconPath(_context.API.GetCurrentTheme());
+        }
+
+        /// <summary>
+        /// Return a filtered list, based on the given query
+        /// </summary>
+        /// <param name="query">The query to filter the list</param>
+        /// <returns>A filtered list, can be empty when nothing was found</returns>
+        public List<Result> Query(Query query)
+        {
+            if (!_oneNoteInstalled || query is null || string.IsNullOrWhiteSpace(query.Search))
+            {
+                return new List<Result>(0);
+            }
+
+            var pages = OneNoteProvider.FindPages(query.Search);
+
+            return pages.Select(p => new Result
+            {
+                IcoPath = _iconPath,
+                Title = p.Name,
+                QueryTextDisplay = p.Name,
+                SubTitle = @$"{p.Notebook.Name}\{p.Section.Name}",
+                Action = (_) =>
+                {
+                    p.OpenInOneNote();
+                    ShowOneNote();
+                    return true;
+                },
+                ContextData = p,
+                ToolTipData = new ToolTipData(Name, @$"{p.Notebook.Name}\{p.Section.Name}\{p.Name}"),
```

Comment:
We could show the file path if possible in the second tool rip line.

Reply:
See other comment